eloquent-javascript/EJS-03.js

```diff
@@ -114,25 +114,23 @@ console.log(nthRecursive(arrayToList([10, 20, 30]), 2));
 
 console.log("\n Deep Equal comparison \n");
 const deepEqual = (objA, objB) => {
+  if (objA === objB) return true;
   if (
-    typeof objA == "object" &&
-    typeof objB == "object" &&
-    objA != null &&
-    objB != null
-  ) {
-    if (Object.keys(objA).length === Object.keys(objB).length) {
-      for (let key of Object.keys(objA)) {
-        if (Object.keys(objB).includes(key)) {
-          if (JSON.stringify(objA[key]) !== JSON.stringify(objB[key])) {
-            return false;
-          }
-        }
-      }
-    }
+    typeof objA != "object" ||
+    objA == null ||
+    typeof objB != "object" ||
+    objB == null
+  )
+    return false;
+  const objKeysA = Object.keys(objA),
+    objKeysB = Object.keys(objB);
+  if (objKeysA.length !== objKeysB.length) return false;
+  for (const key of objKeysA) {
+    if (!objKeysB.includes(key)) return false;
+    return deepEqual(objA[key], objB[key]);
   }
   return true;
 };
-
 console.log('let obj = { here: { is: "an" }, object: 2 };');
 let obj = { here: { is: "an" }, object: 2 };
```

Comment:
Why let? Why not const?

Reply:
That was an example by the book, so I just copied it. And now I changed it to const.